docs/concepts/crds/backupconfiguration_template.md

```diff
@@ -0,0 +1,107 @@
+
+## BackupConfigurationTemplate CRD Specification
+
+Like any official Kubernetes resource, a `BackupConfigurationTemplate` has `TypeMeta`, `ObjectMeta` and `Spec` sections. However, unlike other Kubernetes resources, it does not have a `Status` section.
+
+A sample `BackupConfigurationTemplate` object to backup a Deployment's data through default backup is shown below,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfigurationTemplate
+metadata:
+  name: workload-backup-template
+spec:
+  # ============== Template for Repository ==========================
+  backend:
+    gcs:
+      bucket: stash-backup
+      prefix: stash/${TARGET_NAMESPACE}/${TARGET_KIND}/${TARGET_NAME}
+    storageSecretName: gcs-secret
+  wipeOut: false
+  # ============== Template for BackupConfiguration =================
+  schedule: "* * * * *"
+  # task: # no task section is required for workload data backup
+  #   name: workload-backup
+  runtimeSettings:
+    container:
+      securityContext:
+        runAsUser: 2000
+        runAsGroup: 2000
+  tempDir:
+    medium: "Memory"
+    size:  "1Gi"
+    disableCache: false
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+The sample `BackupConfigurationTemplate` that has been shown above can be used to backup Deployments, DaemonSets, StatefulSets, ReplicaSets and ReplicationControllers. You only have to add some annotations to these workloads. For more details about what annotations you have to add to the targets, please visit [here](/docs/guides/default-backup/overview.md).
+
+Here, we are going to describe some important sections of `BackupConfigurationTemplate` crd.
+
+### BackupConfigurationTemplate `Spec`
+
+We can divide BackupConfigurationTemplate's `.spec` section into two parts. One part specifies template for `Repository` object and other specifies template for `BackupConfiguration` object.
+
+#### Repository Template
+
+You can configure `Repository` template using `spec.backend` field and `spec.wipeOut` field.
+
+- **spec.backend :** `spec.backend` section is backend specification similar to [spec.backend](/docs/concepts/crds/repository.md#specbackend) section of a `Repository` crd. There is only one difference. You can now template `prefix` section (`subPath` for local volume) of the backend to store backed up data of different workloads at different directory. You can use the following variables to template `spec.backend` section:
+
+    |       Variable       |            Usage            |
+    | -------------------- | --------------------------- |
+    | `TARGET_API_VERSION` | API version of the target   |
```

Comment:
- How are we handling upper case vs lower case? 
- TARGET_RESOURCE (plural) should also be available as a variable?

Reply:
- How are we handling upper case vs lower case? 
- TARGET_RESOURCE (plural) should also be available as a variable?

Reply:
https://github.com/kmodules/client-go/blob/master/discovery/restmapper.go#L72